drivers/clock_control/clock_control_nrf2_hsfll.c

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_hsfll
+
+#include "clock_control_nrf2_common.h"
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control/nrf_clock_control.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(clock_control_nrf2, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#include <hal/nrf_hsfll.h>
+
+/* TODO: add support for other HSFLLs */
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "multiple instances not supported");
+
+#include <hal/nrf_hsfll.h>
+
+#ifdef CONFIG_NRFS_HAS_DVFS_SERVICE
+#include <ld_dvfs_handler.h>
+
+/* TODO: get from Kconfig */
+#define UPDATE_TIMEOUT K_MSEC(2)
+
+static const struct clock_options {
+	uint32_t frequency;
+	enum dvfs_frequency_setting setting;
+} clock_options[] = {
+	/* TODO: move to devicetree? */
+	{
+		.frequency = MHZ(64),
```

Comment:
Frequency is one thing. The other is mode (OPEN/CLOSED/BYPASS). Please confirm that mode is skipped intentionally here.